### PR TITLE
Mutation testing with mutmut

### LINKED ISSUE #120: Perform Mutation Testing to Assess the Adequacy of the Test Suite
Even though GatorGrouper has a high-coverage test suite, it would be a good idea for us to assess the quality of the tests through the use of a mutation testing tool. There are several mutation testing tools for Python, such as MutPy and Cosmic-Ray. However, we would need to investigate the types of mutation operators employed by these tools and learn how to introduce them into our build system.

### PR DESCRIPTION
**Pull request by leonardoz15**
# Description of Pull Request

Added `mutmut` for mutation testing to access the current test suite. I also added `mutmut` to the `Pipfile` to install with other dev dependencies.

Fixes #120 

## Type of Change

Please describe the pull request as one of the following: 

- [ ] Bug fix
- [ ] Breaking change
- [x] New feature
- [ ] Documentation update
- [ ] Other 

## Tags

Changes made by @leonardoz15 , @ilikerustoo

### CODE REVIEW

**Comment by leonardoz15** [comment]
Run `mutmut` after installing dev packages with the command `pipenv run mutmut run`
The current output is as follows:

```
- Mutation testing starting - 

These are the steps:
1. A full test suite run will be made to make sure we 
   can run the tests successfully and we know how long 
   it takes (to detect infinite loops for example)
2. Mutants will be generated and checked

Mutants are written to the cache in the .mutmut-cache 
directory. Print found mutants with `mutmut results`.

Legend for output:
🎉 Killed mutants. The goal is for everything to end up in this bucket. 
⏰ Timeout. Test suite took 10 times as long as the baseline so were killed.  
🤔 Suspicious. Tests took a long time, but not long enough to be fatal. 
🙁 Survived. This means your tests needs to be expanded. 

1. Using cached time for baseline tests, to run baseline again delete the cache file

2. Checking mutants
⠼ 330/330  🎉 90  ⏰ 1  🤔 239  🙁 0
```

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[tool:pytest]
+testpaths=gatorgrouper/tests
+
+[mutmut]
+paths_to_mutate=gatorgrouper/
+backup=False
+tests_dir=tests
+runner=python -m pytest
```
Comment (Michionlion) [inline setup.cfg]: Should this mention pipenv? This may be why many mutants are surviving.

Comment (Michionlion) [reply setup.cfg]: @leonardoz15 if this can be fixed or you can confirm it is not a problem, this can be merged!

**Comment by Michionlion** [comment]
@leonardoz15 Please confirm that everything works and fix the conflicts. In the future you need to work in a branch on this repository so that others can access your code.

**Comment by Michionlion** [comment]
@leonardoz15 Your `Pipfile.lock` is corrupted. To regenerate it, you should run `pipenv lock`, and commit the results.

**Pull request merged**